spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -287,18 +287,25 @@ def main(args=None):
     # Straighten spinal cord
     printv('\nStraighten spinal cord...', verbose)
     # check if warp_curve2straight and warp_straight2curve already exist (i.e. no need to do it another time)
+
     cache_sig = cache_signature(
         input_files=[fname_in, fname_seg],
     )
     cachefile = os.path.join(curdir, "straightening.cache")
-    if cache_valid(cachefile, cache_sig) and os.path.isfile(os.path.join(curdir, "warp_curve2straight.nii.gz")) and os.path.isfile(os.path.join(curdir, "warp_straight2curve.nii.gz")) and os.path.isfile(os.path.join(curdir, "straight_ref.nii.gz")):
+    if (cache_valid(cachefile, cache_sig) and
+        os.path.isfile(os.path.join(curdir, "warp_curve2straight.nii.gz")) and 
+        os.path.isfile(os.path.join(curdir, "warp_straight2curve.nii.gz")) and 
+        os.path.isfile(os.path.join(curdir, "straight_ref.nii.gz"))):
+
         # if they exist, copy them into current folder
         printv('Reusing existing warping field which seems to be valid', verbose, 'warning')
         copy(os.path.join(curdir, "warp_curve2straight.nii.gz"), 'warp_curve2straight.nii.gz')
         copy(os.path.join(curdir, "warp_straight2curve.nii.gz"), 'warp_straight2curve.nii.gz')
         copy(os.path.join(curdir, "straight_ref.nii.gz"), 'straight_ref.nii.gz')
         # apply straightening
-        s, o = run_proc(['sct_apply_transfo', '-i', 'data.nii', '-w', 'warp_curve2straight.nii.gz', '-d', 'straight_ref.nii.gz', '-o', 'data_straight.nii'])
+
+        s, o = run_proc(['sct_apply_transfo', '-i', 'data.nii', '-w', 'warp_curve2straight.nii.gz', '-d', 'straight_ref.nii.gz',
```

Comment:
however here we could call the binary directly

Reply:
yes right !